File: ios/Anemone SDK/model/generic/Message.swift
//
//  Message.swift
//  Anemone SDK
//
//  Created by Manuel González Villegas on 14/3/17.
//  Copyright © 2017 Syltek Solutions S.L. All rights reserved.
//

import Foundation

public struct Message {
    public let message: String
    public let status: String

    public init(message: String, status: String) {
        self.message = message
        self.status = status
    }
}

extension Message: JSONMappable {
    public init(json: JSONObject) throws {
        self.message = try json.getString("localized_message")
        self.status = try json.getString("status")
    }
}

extension Message: Equatable { }
public func == (lhs: Message, rhs: Message) -> Bool {
    lhs.status == rhs.status && lhs.message == rhs.message
}

File: android/app/src/main/java/com/playtomic/test/anemone_sdk/model/generic/Message.kt
package com.anemonesdk.model.generic

import com.anemonesdk.general.json.JSONMappable
import com.anemonesdk.general.json.JSONObject
import org.json.JSONException

/**
 * Created by manuelgonzalezvillegas on 14/3/17.
 */

class Message : JSONMappable {

    var message: String

    var status: String

    @Throws(JSONException::class)
    constructor(json: JSONObject) : super(json) {
        this.message = json.getString("localized_message")
        this.status = json.getString("status")
    }

    constructor(message: String, status: String) : super() {
        this.message = message
        this.status = status
    }
}
